There hasn't been an official 3.1 release, but our revision already includes
this typedef.

--- origsrc/qtwinextras-opensource-src-5.2.0/src/winextras/winshobjidl_p.h	2013-12-08 11:10:20.000000000 -0600
+++ src/qtwinextras-opensource-src-5.2.0/src/winextras/winshobjidl_p.h	2014-01-05 15:39:02.804706500 -0600
@@ -222,7 +222,7 @@ public:
 
 #if (defined _MSC_VER && _MSC_VER < 1600) || defined(Q_CC_MINGW)
 
-#   if !defined(__MINGW64_VERSION_MAJOR) || !defined(__MINGW64_VERSION_MINOR) || __MINGW64_VERSION_MAJOR * 100 + __MINGW64_VERSION_MINOR < 301
+#   if !defined(__MINGW64_VERSION_MAJOR) || !defined(__MINGW64_VERSION_MINOR) || __MINGW64_VERSION_MAJOR * 100 + __MINGW64_VERSION_MINOR < 300
 
 typedef struct SHARDAPPIDINFOLINK
 {
@@ -232,7 +232,7 @@ typedef struct SHARDAPPIDINFOLINK
     PCWSTR pszAppID;        // The id of the application that should be associated with this recent doc.
 } SHARDAPPIDINFOLINK;
 
-#   endif // !defined(__MINGW64_VERSION_MAJOR) || !defined(__MINGW64_VERSION_MINOR) || __MINGW64_VERSION_MAJOR * 100 + __MINGW64_VERSION_MINOR < 301
+#   endif // !defined(__MINGW64_VERSION_MAJOR) || !defined(__MINGW64_VERSION_MINOR) || __MINGW64_VERSION_MAJOR * 100 + __MINGW64_VERSION_MINOR < 300
 
 #endif
 
